main/FatesEdgeForestMod.F90

```diff
@@ -0,0 +1,630 @@
+    real(r8), dimension(:), intent(in) :: efb_lognormal_sigmas
+    real(r8), dimension(:), intent(in) :: efb_lognormal_centers
+    real(r8), dimension(:), intent(in) :: efb_quadratic_a
+    real(r8), dimension(:), intent(in) :: efb_quadratic_b
+    real(r8), dimension(:), intent(in) :: efb_quadratic_c
+    real(r8), dimension(:), intent(out) :: fraction_forest_in_bin
+    logical, optional, intent(in) :: norm
+    !
+    ! LOCAL VARIABLES
+    integer :: b  ! Bin index
+    real(r8) :: A      ! Amplitude
+    real(r8) :: mu     ! Center
+    real(r8) :: sigma  ! Sigma
+    logical :: lognorm
+    logical :: do_norm
+    ! Error checking
+    real(r8), parameter :: tol = 1.e-9_r8  ! fraction of total forest area
+    real(r8) :: err_chk
+
+    ! Initialize
+    fraction_forest_in_bin(:) = 0._r8
+
+    ! If the cell is nearly 0% forest, put any forest in the first edge bin (closest to edge)
+    if (x < nearzero) then
+       fraction_forest_in_bin(1) = 1._r8
+       return
+    end if
+
+    ! If the cell is (nearly) 100% forest, it's all "deep forest"
+    if (1._r8 - x < nearzero) then
+       fraction_forest_in_bin(nlevedgeforest) = 1._r8
+       return
+    end if
+
+    if (present(norm)) then
+       do_norm = norm
+    else
+       do_norm = .true.
+    end if
+
+    binloop: do b = 1, nlevedgeforest
+
+       if (is_param_set(efb_gaussian_amplitudes(b)) .or. is_param_set(efb_lognormal_amplitudes(b))) then
+         ! Gaussian or Lognormal
+         lognorm = is_param_set(efb_lognormal_amplitudes(b))
+         if (lognorm) then
+            A = efb_lognormal_amplitudes(b)
+            mu = efb_lognormal_centers(b)
+            sigma = efb_lognormal_sigmas(b)
+         else
+            A = efb_gaussian_amplitudes(b)
+            mu = efb_gaussian_centers(b)
+            sigma = efb_gaussian_sigmas(b)
+         end if
+         fraction_forest_in_bin(b) = GetFracEdgeForestInEachBin_norm(x, A, mu, sigma, lognorm)
+
+       else if (is_param_set(efb_quadratic_a(b))) then
+         ! Quadratic
+         fraction_forest_in_bin(b) = GetFracEdgeForestInEachBin_quadratic(x, efb_quadratic_a(b), efb_quadratic_b(b), efb_quadratic_c(b))
+
+       else
+         call endrun("Unrecognized bin fit type")
+       end if
+    end do binloop
+
+    ! Account for fit errors by normalizing to 1
+    if (do_norm) then
+       fraction_forest_in_bin(:) = fraction_forest_in_bin(:) / sum(fraction_forest_in_bin)
+
+       err_chk = sum(fraction_forest_in_bin) - 1._r8
+       if (abs(err_chk) > tol) then
+          write(fates_log(),*) "ERROR: bin fractions don't sum to 1;    actual minus expected = ",err_chk
+          call endrun(msg=errMsg(__FILE__, __LINE__))
+       end if
+    end if
+
+
+  end subroutine GetFracEdgeForestInEachBin
+
+
+  subroutine AssignPatchToBins(fraction_forest_in_each_bin, area_forest_patches, patch_area, nlevedgeforest, tol, sum_forest_bins_so_far_m2, area_in_edgeforest_bins_m2)
```

Comment:
I see the differences between subroutines `AssignPatchToBins` and `AssignPatchesToBins`, but it is a bit confusing. Perhaps rename one of them to something a bit more unique, for example, rename the latter to `DistributePatchesAcrossBins` or something like this?